pydmd/dmdbase.py

```diff
@@ -672,13 +686,19 @@ def _optimal_dmd_matrices(self):
                 )
             )
         else:
-            _, s, V = compute_svd(self.snapshots[:, :-1], self.modes.shape[-1])
+            if self._snapshots_holder_y:
+                X = self.snapshots
+            else:
+                X = self.snapshots[:, :-1]
+
+            _, s, V = compute_svd(X, self.modes.shape[-1])
+            n = X.shape[1]
 
             q = np.conj(
                 np.diag(
                     np.linalg.multi_dot(
                         [
-                            vander[:, :-1],
+                            vander[:, :n],
```

Comment:
`n` is only used once:
```suggestion
                            vander[:, :X.shape[1]],
```

If you apply this, please delete the definition of `n` above.

Reply:
Done!